service/history/workflowExecutionContext.go

```diff
@@ -583,6 +583,23 @@ func (c *workflowExecutionContextImpl) update(transferTasks []persistence.Task,
 					return err
 				}
 				executionInfo.SetLastFirstEventID(firstEvent.GetEventId())
+
+				// add clean up tasks
+				transferTasks = append(transferTasks, &persistence.CloseExecutionTask{})
```

Comment:
I think we already have a helper method in history engine which generates relevant tasks on workflow completion.  I think we should use the same helper method or create one if it does not exist.  These tasks need to be generated everytime workflow execution completes so we need to make sure this logic remains consistent everywhere.

Reply:
agree, addressed.